File: Android/app/src/main/java/com/example/snaptaplaque/models/api/favorites/FavoriteAllResponse.java
package com.example.snaptaplaque.models.api.favorites;

import com.example.snaptaplaque.models.api.vehicles.InfoResponse;

import java.util.List;

/**
 * Modèle de données représentant la réponse de l'API lors de la récupération de tous les favoris.
 * * <p>Cette classe encapsule une liste d'objets {@link InfoResponse}, permettant de
 * centraliser les informations techniques de tous les véhicules que l'utilisateur
 * a choisi de sauvegarder dans sa bibliothèque personnelle.</p>
 */
public class FavoriteAllResponse {

    /** * Liste contenant l'ensemble des véhicules marqués comme favoris.
     */
    private List<InfoResponse> favorites;

    /**
     * Constructeur pour initialiser la réponse avec la liste des favoris.
     *
     * @param favorites La liste des objets {@link InfoResponse} récupérée depuis le serveur.
     */
    public FavoriteAllResponse(List<InfoResponse> favorites) { this.favorites = favorites; }

    /**
     * Récupère la liste complète des véhicules favoris.
     *
     * @return Une {@link List} d'objets {@link InfoResponse} représentant les favoris de l'utilisateur.
     */
    public List<InfoResponse> getFavorites() { return favorites; }
}

File: iOS/SnapTaPlaque/Models/API/Favorites/FavoriteAllResponse.swift
//
//  FavoriteAllResponse.swift
//  SnapTaPlaque
//
//  Created by Victor Poirier on 20/03/2026.
//

import Foundation

/// Représente la réponse de l'API lorsqu'on récupère la liste complète des véhicules favoris.
///
/// Cette structure se conforme au protocole `Codable` afin de traduire un retour JSON
/// (contenant typiquement un tableau à la racine sous la clé `favorites`) depuis le backend
/// vers un modèle de données natif exploitable dans les vues SwiftUI de "SnapTaPlaque".
struct FavoriteAllResponse: Codable {
    
    /// Un tableau contenant les informations détaillées de chacun des véhicules marqués en favoris.
    ///
    /// - Note: Ce tableau s'appuie sur le modèle `InfoResponse` pour la structure de chaque élément.
    let favorites: [InfoResponse]
}
